### DISCUSSION TITLE
Enable JSON Templating with Contact Point Webhooks to expand integrations

### DISCUSSION THREAD

**Discussion by kylebrandt** [grafana/grafana#38600] [Alerting]
In order to support more generic JSON/REST endpoints as contact points, I propose we add templating to the webhook contact point type (or a new different contact point type, but similar one.)

There is a long tail of things people want to use as contact points, but limited bandwidth to support various contact points integrations. By enabling templating with some extra functions, users will be able to make and share templates that work with various systems.

These extra functions are something like:
 - `makeSlice` (and `append`)
 -  `makeMap`
 -  `json`
 
These enabled users to make a template that builds an object (made up of arrays(slices) and dictionaries (maps)) and then use the `json` function to serialize it. (`json` would be mapped to `json.Marhsal` (So users don't have to escape strings or anything like that)).

In a previous system I worked on it looked like the following:

```
{{- $html_link := printf "<a href=\"%v\">#%v</a>" .Incident .Id -}}
{{- $text := printf "Alert %v is in %v state, see %v for details" .AlertKey .CurrentStatus $html_link -}}
{{- makeMap "key" (V "$apiToken") "incident" .Id "description" (makeSlice .Subject $text) | json -}}
```

So the templates can get pretty gnarly. But since they are reusable, I think it can save users form having to use intermediaries to integrate with various systems. Also authors of third party systems would be able to include such templates in their documentation.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
